docker/src/main/java/clocker/docker/location/strategy/basic/GroupPlacementStrategy.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2014-2016 by Cloudsoft Corporation Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package clocker.docker.location.strategy.basic;
+
+import static clocker.docker.location.strategy.util.StrategyPredicates.childrenOf;
+import static clocker.docker.location.strategy.util.StrategyPredicates.hasApplicationId;
+import static clocker.docker.location.strategy.util.StrategyPredicates.nonEmpty;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import clocker.docker.location.DockerHostLocation;
+import clocker.docker.location.strategy.AbstractDockerPlacementStrategy;
+import clocker.docker.location.strategy.DockerAwarePlacementStrategy;
+
+import com.google.common.annotations.Beta;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.reflect.TypeToken;
+import com.google.common.util.concurrent.Monitor;
+import com.google.common.util.concurrent.Uninterruptibles;
+
+import org.apache.brooklyn.api.entity.Entity;
+import org.apache.brooklyn.config.ConfigKey;
+import org.apache.brooklyn.core.config.ConfigKeys;
+import org.apache.brooklyn.util.collections.MutableList;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.core.flags.SetFromFlag;
+import org.apache.brooklyn.util.core.mutex.WithMutexes;
+import org.apache.brooklyn.util.exceptions.Exceptions;
+import org.apache.brooklyn.util.time.Duration;
+
+/**
+ * A {@link DockerAwarePlacementStrategy strategy} that requires entities with
+ * the same parent to use the same host.
+ *
+ * Can be configured to require exclusive use of the host with the
+ * {@link #REQUIRE_EXCLUSIVE exclusive} ({@code docker.constraint.exclusive})
+ * option set to {@code true}; normally {@code false}.
+ *
+ * @since 1.1.0
+ */
+@Beta
+@ThreadSafe
+public class GroupPlacementStrategy extends AbstractDockerPlacementStrategy implements WithMutexes {
```

Comment:
I wonder if `GroupPlacementStrategy` is the best name. Perhaps `SiblingPlacementStrategy`? "Group" is a very generic term, and in Brooklyn has a more specific meaning that is not applicable here.